src/Bicep.LangServer/Handlers/BicepTextDocumentSyncHandler.cs

```diff
@@ -34,30 +48,68 @@ public override Task<Unit> Handle(DidChangeTextDocumentParams request, Cancellat
             // we have full sync enabled, so apparently first change is the whole document
             var contents = request.ContentChanges.First().Text;
 
-            this.compilationManager.UpsertCompilation(request.TextDocument.Uri, request.TextDocument.Version, contents);
+            var documentUri = request.TextDocument.Uri;
+
+            this.compilationManager.UpsertCompilation(documentUri, request.TextDocument.Version, contents);
+
+            if (IsBicepConfigFile(documentUri))
+            {
+                var configuration = configurationManager.GetConfiguration(documentUri.ToUri());
+                activeBicepConfigCache.AddOrUpdate(documentUri, (documentUri) => configuration, (documentUri, prevConfiguration) => configuration);
+            }
 
             return Unit.Task;
         }
 
         public override Task<Unit> Handle(DidOpenTextDocumentParams request, CancellationToken cancellationToken)
         {
-            this.compilationManager.UpsertCompilation(request.TextDocument.Uri, request.TextDocument.Version, request.TextDocument.Text, request.TextDocument.LanguageId);
+            var documentUri = request.TextDocument.Uri;
+
+            if (IsBicepConfigFile(documentUri))
+            {
+                var configuration = configurationManager.GetConfiguration(documentUri.ToUri());
```

Comment:
Can this throw? What happens if it does?

Reply:
It can throw when the config file is invalid or if we get access denied. If an exception is thrown here, the compilation won't be updated with the open file and the file will probably be permanently busted in vs code.

Reply:
Added error handling to not block compilation:
https://github.com/Azure/bicep/commit/979995aec3298bb62719b50267e746b3fd667ffb